src/Util/Network.php

```diff
@@ -337,6 +327,21 @@ public static function post(string $url, $params, array $headers = [], int $time
 
 		$a->getProfiler()->saveTimestamp($stamp1, 'network', System::callstack());
 
+		// Very old versions of Lighttpd don't like the "Expect" header, so we remove it when needed
+		if ($curlResponse->getReturnCode() == 417) {
+			$redirects++;
+
+			if (empty($headers)) {
+				$headers = ['Expect:'];
+			} else {
+				if (!in_array('Expect:', $headers)) {
```

Comment:
This will miss a fully formed `Expect` header like `Expect: application/json` and add an additional `Expect:` header, is this intended?

Reply:
Don't mix the headers :-) See here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect

Reply:
Fair enough, same question with `Expect: 100-Continue`?

Reply:
Fair enough, same question with `Expect: 100-Continue`?

Reply:
Especially this header is creating the "417" HTTP error on older versions of that web server, so we have to get rid of it.

Reply:
How do you get rid of it by adding it?

Reply:
How are you getting rid of it by adding it?

Reply:
Setting it to an empty value actually removes it.